pkg/cfn/builder/cluster_test.go

```diff
@@ -172,34 +172,65 @@ var _ = Describe("Cluster Template Builder", func() {
 			Expect(clusterTemplate.Resources).To(HaveKey("ClusterSharedNodeSecurityGroup"))
 		})
 
-		Context("when 1 extraCIDR is defined", func() {
+		Context("when extraCIDRs are defined", func() {
 			BeforeEach(func() {
-				oneExtraCIDR := []string{"192.168.0.0/24"}
-				cfg.VPC.ExtraCIDRs = oneExtraCIDR
+				cfg.VPC.ExtraCIDRs = []string{"192.168.0.0/24", "192.168.1.0/24"}
 			})
 
-			It("should add 1 extra control plane ingress rule", func() {
+			It("should add extra control plane ingress rules", func() {
 				Expect(clusterTemplate.Resources).To(HaveKey("IngressControlPlaneExtraCIDR0"))
-				Expect(clusterTemplate.Resources["IngressControlPlaneExtraCIDR0"].Properties.CidrIP).To(Equal("192.168.0.0/24"))
-				Expect(clusterTemplate.Resources["IngressControlPlaneExtraCIDR0"].Properties.IPProtocol).To(Equal("tcp"))
-				Expect(clusterTemplate.Resources["IngressControlPlaneExtraCIDR0"].Properties.FromPort).To(Equal(443))
-				Expect(clusterTemplate.Resources["IngressControlPlaneExtraCIDR0"].Properties.ToPort).To(Equal(443))
+
+				Expect(clusterTemplate.Resources["IngressControlPlaneExtraCIDR0"].Properties).To(Equal(fakes.Properties{
+					CidrIP:     "192.168.0.0/24",
+					IPProtocol: "tcp",
+					FromPort:   443,
+					ToPort:     443,
+					GroupID: map[string]interface{}{
+						"Ref": "ControlPlaneSecurityGroup"},
+					Description: "Allow Extra CIDR 0 (192.168.0.0/24) to communicate to controlplane",
+				}))
+
+				Expect(clusterTemplate.Resources).To(HaveKey("IngressControlPlaneExtraCIDR1"))
+				Expect(clusterTemplate.Resources["IngressControlPlaneExtraCIDR1"].Properties).To(Equal(fakes.Properties{
+					CidrIP:     "192.168.1.0/24",
+					IPProtocol: "tcp",
+					FromPort:   443,
+					ToPort:     443,
+					GroupID: map[string]interface{}{
+						"Ref": "ControlPlaneSecurityGroup"},
+					Description: "Allow Extra CIDR 1 (192.168.1.0/24) to communicate to controlplane",
+				}))
```

Comment:
? Just a precaution that multiple rules can be defined?

Reply:
got to have a test for the for-loop :wink: